ui_testing/testcases/basic_tests/test05_testplans.py

```diff
@@ -51,8 +51,417 @@ def test001_test_plan_delete_testunit(self):
+                                            article=second_article_data['name'])
+        self.base_selenium.LOGGER.info(
+            'New testplan is created successfully with name: {}, article name: {} and material type: {}'.format(
+                testplan_name, second_article_data['name'], second_article_data['materialType']))
+
+        data = self.test_plan.search(testplan_name)
+        search_length = 0
+        for d in data:
+            if len(d.text) != 0:
+                search_length += 1
+
+        self.assertEqual(search_length, 2)
+
+    def test012_create_testplans_same_name_materialtype_all_article(self):
+        '''
+        LIMS-3500
+        New: Test plan: Creation Approach: I can't create two test plans
+        with the same name & same materiel type & one with any article
+        and the other one all
+        '''
+
+        # get all articles and choose a random one to take its information
+        articles = self.get_all_articles()
+        random_index = self.generate_random_number(lower=0, upper=len(articles))
+        article_data = articles[random_index]
+        self.base_selenium.LOGGER.info(
+            'A random article is chosen, its name: {} and its material type: {}'.format(article_data['name'],
+                                                                                        article_data[
+                                                                                            'materialType']))
+
+        testplan_name = self.test_plan.create_new_test_plan(material_type=article_data['materialType'],
+                                                            article=article_data['name'])
+        self.base_selenium.LOGGER.info(
+            'New testplan is created successfully with name: {}, article name: {} and material type: {}'.format(
+                testplan_name, article_data['name'], article_data['materialType']))
+
+        self.base_selenium.LOGGER.info(
+            'Attempting to create another testplan with the same name & material type as the previously created one,'
+            ' and all articles')
+
+        # create another testplan with the same data
+        self.test_plan.create_new_test_plan(name=testplan_name, material_type=article_data['materialType'],
+                                            article='All')
+
+        self.base_selenium.LOGGER.info(
+            'Waiting for the error message to make sure that validation forbids the creation of two testplans having the same name, material type and article')
+        validation_result = self.base_selenium.wait_element(element='general:oh_snap_msg')
 
+        self.base_selenium.LOGGER.info(
+            'Assert the error message to make sure that validation forbids the creation of two testplans having the same name, material type one of any article and the other for all articles? {}'.format(
+                validation_result))
+        self.assertTrue(validation_result)
```

Comment:
assert the validation_result.text with the expected message

Reply:
validation_result has no element called text, it's a boolean